### DISCUSSION TITLE
Terminal Dormancy Issue

### DISCUSSION THREAD

**Discussion by maeryo** [coder/code-server#6390] [Ideas] [answered]
When running a task in the terminal in the foreground and closing the browser, the ongoing process is temporarily paused. Reopening the browser and reconnecting the terminal resumes the paused process.

To address this, I installed GNU's screen and configured both bashrc and screenrc. As a result, whenever a new terminal is opened, a new screen session starts, and when that terminal is closed, the screen session is terminated automatically.

However, this is merely a user-specific customization. To fundamentally solve this issue, the terminal connection method in code-server needs improvement. Alternatively, a combined approach using bash and screen, similar to what I implemented, could be adopted.

I'd love to hear your thoughts on this.

Related issue and discussion:
https://github.com/coder/code-server/issues/5589
https://github.com/coder/code-server/discussions/5590

**Comment by code-asher** [comment, marked as answer]
I think any fundamental changes to the terminals would need to go upstream (https://github.com/microsoft/vscode), patching them in code-server would probably be difficult.  At least, this sounds to me like a non-trivial change.
